common/numerated/src/tree.rs

```diff
@@ -417,7 +418,7 @@ impl<T: Numerated> IntervalsTree<T> {
     pub fn points_iter(&self) -> impl Iterator<Item = T> + '_ {
```

Comment:
de-facto you copy everything, isn't it better to wrap map iterator and expand point lazy: on `.next()` called?

Reply:
> de-facto you copy everything
would be grate to see some proofs